modules/ducktests/tests/ignitetest/tests/control_utility_test.py

```diff
@@ -0,0 +1,235 @@
+        new_node = self.__start_ignite_nodes(version, 1)
+        _, version, _ = control_utility.cluster_state()
+        control_utility.set_baseline(version)
+        blt_size += 1
+
+        baseline = control_utility.baseline()
+        self.__check_baseline_size(baseline, blt_size)
+        self.__check_nodes_in_baseline(new_node.nodes, baseline)
+
+    @cluster(num_nodes=NUM_NODES)
+    @parametrize(version=str(DEV_BRANCH))
+    @parametrize(version=str(LATEST_2_8))
+    @parametrize(version=str(LATEST_2_7))
+    def test_baseline_add_remove(self, version):
+        """
+        Test add and remove nodes from baseline.
+        """
+        blt_size = self.NUM_NODES - 1
+        self.servers = self.__start_ignite_nodes(version, blt_size)
+
+        control_utility = ControlUtility(self.servers, self.test_context)
+
+        control_utility.activate()
+
+        # Add node to baseline.
+        new_node = self.__start_ignite_nodes(version, 1)
+        control_utility.add_to_baseline(new_node.nodes)
+        blt_size += 1
+
+        baseline = control_utility.baseline()
+        self.__check_baseline_size(baseline, blt_size)
+        self.__check_nodes_in_baseline(new_node.nodes, baseline)
+
+        # Expected failure (remove of online node is not allowed).
+        try:
+            control_utility.remove_from_baseline(new_node.nodes)
+
+            assert False, "Remove of online node from baseline should fail!"
+        except ControlUtilityError:
+            pass
+
+        # Remove of offline node from baseline.
+        new_node.stop()
+
+        self.servers.await_event("Node left topology", timeout_sec=30, from_the_beginning=True)
+
+        control_utility.remove_from_baseline(new_node.nodes)
+        blt_size -= 1
+
+        baseline = control_utility.baseline()
+        self.__check_baseline_size(baseline, blt_size)
+        self.__check_nodes_not_in_baseline(new_node.nodes, baseline)
+
+    @cluster(num_nodes=NUM_NODES)
+    @parametrize(version=str(DEV_BRANCH))
+    @parametrize(version=str(LATEST_2_8))
+    @parametrize(version=str(LATEST_2_7))
+    def test_activate_deactivate(self, version):
+        """
+        Test activate and deactivate cluster.
+        """
+        self.servers = self.__start_ignite_nodes(version, self.NUM_NODES)
+
+        control_utility = ControlUtility(self.servers, self.test_context)
+
+        control_utility.activate()
+
+        state, _, _ = control_utility.cluster_state()
+
+        assert state.lower() == 'active', 'Unexpected state %s' % state
+
+        control_utility.deactivate()
+
+        state, _, _ = control_utility.cluster_state()
+
+        assert state.lower() == 'inactive', 'Unexpected state %s' % state
+
+    @cluster(num_nodes=NUM_NODES)
+    @parametrize(version=str(DEV_BRANCH))
+    @parametrize(version=str(LATEST_2_8))
+    def test_baseline_autoadjust(self, version):
+        """
+        Test activate and deactivate cluster.
+        """
+        if version < V_2_8_0:
```

Comment:
This condition is not part of test itself, let's make decorator from that. Otherwise we should not used it at all, as every test should be marked this way.

Reply:
@timoninmaxim could explain how to do this properly in detail, if possible.

Reply:
@anton-vinogradov See my next PR #8137